# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1454528233 18000
#      Wed Feb 03 14:37:13 2016 -0500
# Node ID a65b906e25bc5c9a7871cc659d012b64972cd7dd
# Parent  a428c6ba63344734b9ea045928738aeb62af2740
revamp oct-conf.h

* oct-conf.in.cc: New file.  Use character strings in a namespace
instead of macros to define configuration info.  Omit configuration
info beginning with ENABLE_ since those appear as macros in config.h
* oct.conf.h: New file.
* oct-conf.in.h: Delete.
* config-features.sh: Also recognize macros beginning with ENABLE_
* toplev.cc (octave_config_info): Update to match changes in
oct-conf.h and oct-conf-features.h.
* __init_gnuplot__.cc (have_gnuplot_binar): Likewise.
* libinterp/module.mk: Update for new files.

diff --git a/libinterp/config-features.sh b/libinterp/config-features.sh
--- a/libinterp/config-features.sh
+++ b/libinterp/config-features.sh
@@ -14,22 +14,22 @@ cat << EOF
 octave_scalar_map
 octave_config_features (void)
 {
   octave_scalar_map m;
 
 EOF
 
 $AWK \
-  '/#define HAVE_/ {
-     sub (/HAVE_/, "", $2);
+  '/#define (HAVE|ENABLE)_/ {
+     sub (/(HAVE|ENABLE)_/, "", $2);
      printf ("  m.assign (\"%s\", octave_value (true));\n", $2);
    }
-   /\/\* #undef HAVE_/ {
-     sub (/HAVE_/, "", $3);
+   /\/\* #undef (HAVE|ENABLE)_/ {
+     sub (/(HAVE|ENABLE)_/, "", $3);
      printf ("  m.assign (\"%s\", octave_value (false));\n", $3);
    } {
    }' $conffile
 
 cat << EOF
 
   return m;
 }
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1256,170 +1256,162 @@ specified option.\n\
   {
     bool subst_home;
     const char *key;
     const char *val;
   };
 
   static const conf_info_struct conf_info[] =
   {
-    { false, "ALL_CFLAGS", OCTAVE_CONF_ALL_CFLAGS },
-    { false, "ALL_CXXFLAGS", OCTAVE_CONF_ALL_CXXFLAGS },
-    { false, "ALL_FFLAGS", OCTAVE_CONF_ALL_FFLAGS },
-    { false, "ALL_LDFLAGS", OCTAVE_CONF_ALL_LDFLAGS },
-    { false, "AMD_CPPFLAGS", OCTAVE_CONF_AMD_CPPFLAGS },
-    { false, "AMD_LDFLAGS", OCTAVE_CONF_AMD_LDFLAGS },
-    { false, "AMD_LIBS", OCTAVE_CONF_AMD_LIBS },
-    { false, "AR", OCTAVE_CONF_AR },
-    { false, "ARFLAGS", OCTAVE_CONF_ARFLAGS },
-    { false, "ARPACK_CPPFLAGS", OCTAVE_CONF_ARPACK_CPPFLAGS },
-    { false, "ARPACK_LDFLAGS", OCTAVE_CONF_ARPACK_LDFLAGS },
-    { false, "ARPACK_LIBS", OCTAVE_CONF_ARPACK_LIBS },
-    { false, "BLAS_LIBS", OCTAVE_CONF_BLAS_LIBS },
-    { false, "CAMD_CPPFLAGS", OCTAVE_CONF_CAMD_CPPFLAGS },
-    { false, "CAMD_LDFLAGS", OCTAVE_CONF_CAMD_LDFLAGS },
-    { false, "CAMD_LIBS", OCTAVE_CONF_CAMD_LIBS },
-    { false, "CARBON_LIBS", OCTAVE_CONF_CARBON_LIBS },
-    { false, "CC", OCTAVE_CONF_CC },
-    { false, "CCOLAMD_CPPFLAGS", OCTAVE_CONF_CCOLAMD_CPPFLAGS },
-    { false, "CCOLAMD_LDFLAGS", OCTAVE_CONF_CCOLAMD_LDFLAGS },
-    { false, "CCOLAMD_LIBS", OCTAVE_CONF_CCOLAMD_LIBS },
-    { false, "CFLAGS", OCTAVE_CONF_CFLAGS },
-    { false, "CHOLMOD_CPPFLAGS", OCTAVE_CONF_CHOLMOD_CPPFLAGS },
-    { false, "CHOLMOD_LDFLAGS", OCTAVE_CONF_CHOLMOD_LDFLAGS },
-    { false, "CHOLMOD_LIBS", OCTAVE_CONF_CHOLMOD_LIBS },
-    { false, "COLAMD_CPPFLAGS", OCTAVE_CONF_COLAMD_CPPFLAGS },
-    { false, "COLAMD_LDFLAGS", OCTAVE_CONF_COLAMD_LDFLAGS },
-    { false, "COLAMD_LIBS", OCTAVE_CONF_COLAMD_LIBS },
-    { false, "CPICFLAG", OCTAVE_CONF_CPICFLAG },
-    { false, "CPPFLAGS", OCTAVE_CONF_CPPFLAGS },
-    { false, "CURL_CPPFLAGS", OCTAVE_CONF_CURL_CPPFLAGS },
-    { false, "CURL_LDFLAGS", OCTAVE_CONF_CURL_LDFLAGS },
-    { false, "CURL_LIBS", OCTAVE_CONF_CURL_LIBS },
-    { false, "CXSPARSE_CPPFLAGS", OCTAVE_CONF_CXSPARSE_CPPFLAGS },
-    { false, "CXSPARSE_LDFLAGS", OCTAVE_CONF_CXSPARSE_LDFLAGS },
-    { false, "CXSPARSE_LIBS", OCTAVE_CONF_CXSPARSE_LIBS },
-    { false, "CXX", OCTAVE_CONF_CXX },
-    { false, "CXXCPP", OCTAVE_CONF_CXXCPP },
-    { false, "CXXFLAGS", OCTAVE_CONF_CXXFLAGS },
-    { false, "CXXPICFLAG", OCTAVE_CONF_CXXPICFLAG },
+    { false, "ALL_CFLAGS", octave::config::ALL_CFLAGS },
+    { false, "ALL_CXXFLAGS", octave::config::ALL_CXXFLAGS },
+    { false, "ALL_FFLAGS", octave::config::ALL_FFLAGS },
+    { false, "ALL_LDFLAGS", octave::config::ALL_LDFLAGS },
+    { false, "AMD_CPPFLAGS", octave::config::AMD_CPPFLAGS },
+    { false, "AMD_LDFLAGS", octave::config::AMD_LDFLAGS },
+    { false, "AMD_LIBS", octave::config::AMD_LIBS },
+    { false, "AR", octave::config::AR },
+    { false, "ARFLAGS", octave::config::ARFLAGS },
+    { false, "ARPACK_CPPFLAGS", octave::config::ARPACK_CPPFLAGS },
+    { false, "ARPACK_LDFLAGS", octave::config::ARPACK_LDFLAGS },
+    { false, "ARPACK_LIBS", octave::config::ARPACK_LIBS },
+    { false, "BLAS_LIBS", octave::config::BLAS_LIBS },
+    { false, "CAMD_CPPFLAGS", octave::config::CAMD_CPPFLAGS },
+    { false, "CAMD_LDFLAGS", octave::config::CAMD_LDFLAGS },
+    { false, "CAMD_LIBS", octave::config::CAMD_LIBS },
+    { false, "CARBON_LIBS", octave::config::CARBON_LIBS },
+    { false, "CC", octave::config::CC },
+    { false, "CCOLAMD_CPPFLAGS", octave::config::CCOLAMD_CPPFLAGS },
+    { false, "CCOLAMD_LDFLAGS", octave::config::CCOLAMD_LDFLAGS },
+    { false, "CCOLAMD_LIBS", octave::config::CCOLAMD_LIBS },
+    { false, "CFLAGS", octave::config::CFLAGS },
+    { false, "CHOLMOD_CPPFLAGS", octave::config::CHOLMOD_CPPFLAGS },
+    { false, "CHOLMOD_LDFLAGS", octave::config::CHOLMOD_LDFLAGS },
+    { false, "CHOLMOD_LIBS", octave::config::CHOLMOD_LIBS },
+    { false, "COLAMD_CPPFLAGS", octave::config::COLAMD_CPPFLAGS },
+    { false, "COLAMD_LDFLAGS", octave::config::COLAMD_LDFLAGS },
+    { false, "COLAMD_LIBS", octave::config::COLAMD_LIBS },
+    { false, "CPICFLAG", octave::config::CPICFLAG },
+    { false, "CPPFLAGS", octave::config::CPPFLAGS },
+    { false, "CURL_CPPFLAGS", octave::config::CURL_CPPFLAGS },
+    { false, "CURL_LDFLAGS", octave::config::CURL_LDFLAGS },
+    { false, "CURL_LIBS", octave::config::CURL_LIBS },
+    { false, "CXSPARSE_CPPFLAGS", octave::config::CXSPARSE_CPPFLAGS },
+    { false, "CXSPARSE_LDFLAGS", octave::config::CXSPARSE_LDFLAGS },
+    { false, "CXSPARSE_LIBS", octave::config::CXSPARSE_LIBS },
+    { false, "CXX", octave::config::CXX },
+    { false, "CXXCPP", octave::config::CXXCPP },
+    { false, "CXXFLAGS", octave::config::CXXFLAGS },
+    { false, "CXXPICFLAG", octave::config::CXXPICFLAG },
     { false, "DEFAULT_PAGER", OCTAVE_DEFAULT_PAGER },
-    { false, "DEFS", OCTAVE_CONF_DEFS },
-    { false, "DL_LD", OCTAVE_CONF_DL_LD },
-    { false, "DL_LDFLAGS", OCTAVE_CONF_DL_LDFLAGS },
-    { false, "DL_LIBS", OCTAVE_CONF_DL_LIBS },
-    { false, "GCC_VERSION", OCTAVE_CONF_GCC_VERSION },
-    { false, "GXX_VERSION", OCTAVE_CONF_GXX_VERSION },
-    { false, "ENABLE_64", OCTAVE_CONF_ENABLE_64 },
-    { false, "ENABLE_BOUNDS_CHECK", OCTAVE_CONF_ENABLE_BOUNDS_CHECK },
-    { false, "ENABLE_DOCS", OCTAVE_CONF_ENABLE_DOCS },
-    { false, "ENABLE_DYNAMIC_LINKING", OCTAVE_CONF_ENABLE_DYNAMIC_LINKING },
-    { false, "ENABLE_FLOAT_TRUNCATE", OCTAVE_CONF_ENABLE_FLOAT_TRUNCATE },
-    { false, "ENABLE_JIT", OCTAVE_CONF_ENABLE_JIT },
-    { false, "ENABLE_OPENMP", OCTAVE_CONF_ENABLE_OPENMP },
-    { false, "EXEEXT", OCTAVE_CONF_EXEEXT },
-    { false, "F77", OCTAVE_CONF_F77 },
-    { false, "F77_FLOAT_STORE_FLAG", OCTAVE_CONF_F77_FLOAT_STORE_FLAG },
-    { false, "F77_INTEGER_8_FLAG", OCTAVE_CONF_F77_INTEGER_8_FLAG },
-    { false, "FFLAGS", OCTAVE_CONF_FFLAGS },
-    { false, "FFTW3_CPPFLAGS", OCTAVE_CONF_FFTW3_CPPFLAGS },
-    { false, "FFTW3_LDFLAGS", OCTAVE_CONF_FFTW3_LDFLAGS },
-    { false, "FFTW3_LIBS", OCTAVE_CONF_FFTW3_LIBS },
-    { false, "FFTW3F_CPPFLAGS", OCTAVE_CONF_FFTW3F_CPPFLAGS },
-    { false, "FFTW3F_LDFLAGS", OCTAVE_CONF_FFTW3F_LDFLAGS },
-    { false, "FFTW3F_LIBS", OCTAVE_CONF_FFTW3F_LIBS },
-    { false, "FLIBS", OCTAVE_CONF_FLIBS },
-    { false, "FLTK_CPPFLAGS", OCTAVE_CONF_FLTK_CPPFLAGS },
-    { false, "FLTK_LDFLAGS", OCTAVE_CONF_FLTK_LDFLAGS },
-    { false, "FLTK_LIBS", OCTAVE_CONF_FLTK_LIBS },
-    { false, "FONTCONFIG_CPPFLAGS", OCTAVE_CONF_FONTCONFIG_CPPFLAGS },
-    { false, "FONTCONFIG_LIBS", OCTAVE_CONF_FONTCONFIG_LIBS },
-    { false, "FPICFLAG", OCTAVE_CONF_FPICFLAG },
-    { false, "FT2_CPPFLAGS", OCTAVE_CONF_FT2_CPPFLAGS },
-    { false, "FT2_LIBS", OCTAVE_CONF_FT2_LIBS },
-    { false, "GLPK_CPPFLAGS", OCTAVE_CONF_GLPK_CPPFLAGS },
-    { false, "GLPK_LDFLAGS", OCTAVE_CONF_GLPK_LDFLAGS },
-    { false, "GLPK_LIBS", OCTAVE_CONF_GLPK_LIBS },
-    { false, "GNUPLOT", OCTAVE_CONF_GNUPLOT },
-    { false, "HDF5_CPPFLAGS", OCTAVE_CONF_HDF5_CPPFLAGS },
-    { false, "HDF5_LDFLAGS", OCTAVE_CONF_HDF5_LDFLAGS },
-    { false, "HDF5_LIBS", OCTAVE_CONF_HDF5_LIBS },
-    { false, "LAPACK_LIBS", OCTAVE_CONF_LAPACK_LIBS },
-    { false, "LDFLAGS", OCTAVE_CONF_LDFLAGS },
-    { false, "LD_CXX", OCTAVE_CONF_LD_CXX },
-    { false, "LD_STATIC_FLAG", OCTAVE_CONF_LD_STATIC_FLAG },
-    { false, "LEX", OCTAVE_CONF_LEX },
-    { false, "LEXLIB", OCTAVE_CONF_LEXLIB },
-    { false, "LFLAGS", OCTAVE_CONF_LFLAGS },
-    { false, "LIBEXT", OCTAVE_CONF_LIBEXT },
-    { false, "LIBFLAGS", OCTAVE_CONF_LIBFLAGS },
-    { false, "LIBOCTAVE", OCTAVE_CONF_LIBOCTAVE },
-    { false, "LIBOCTINTERP", OCTAVE_CONF_LIBOCTINTERP },
-    { false, "LIBS", OCTAVE_CONF_LIBS },
-    { false, "LLVM_CPPFLAGS", OCTAVE_CONF_LLVM_CPPFLAGS },
-    { false, "LLVM_LDFLAGS", OCTAVE_CONF_LLVM_LDFLAGS },
-    { false, "LLVM_LIBS", OCTAVE_CONF_LLVM_LIBS },
-    { false, "LN_S", OCTAVE_CONF_LN_S },
-    { false, "MAGICK_CPPFLAGS", OCTAVE_CONF_MAGICK_CPPFLAGS },
-    { false, "MAGICK_LDFLAGS", OCTAVE_CONF_MAGICK_LDFLAGS },
-    { false, "MAGICK_LIBS", OCTAVE_CONF_MAGICK_LIBS },
-    { false, "MKOCTFILE_DL_LDFLAGS", OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS },
-    { false, "OCTAVE_LINK_DEPS", OCTAVE_CONF_OCTAVE_LINK_DEPS },
-    { false, "OCTAVE_LINK_OPTS", OCTAVE_CONF_OCTAVE_LINK_OPTS },
-    { false, "OCT_LINK_DEPS", OCTAVE_CONF_OCT_LINK_DEPS },
-    { false, "OCT_LINK_OPTS", OCTAVE_CONF_OCT_LINK_OPTS },
-    { false, "OPENGL_LIBS", OCTAVE_CONF_OPENGL_LIBS },
-    { false, "OSMESA_CPPFLAGS", OCTAVE_CONF_OSMESA_CPPFLAGS },
-    { false, "OSMESA_LDFLAGS", OCTAVE_CONF_OSMESA_LDFLAGS },
-    { false, "OSMESA_LIBS", OCTAVE_CONF_OSMESA_LIBS },
-    { false, "PCRE_CPPFLAGS", OCTAVE_CONF_PCRE_CPPFLAGS },
-    { false, "PCRE_LIBS", OCTAVE_CONF_PCRE_LIBS },
-    { false, "PTHREAD_CFLAGS", OCTAVE_CONF_PTHREAD_CFLAGS },
-    { false, "PTHREAD_LIBS", OCTAVE_CONF_PTHREAD_LIBS },
-    { false, "QHULL_CPPFLAGS", OCTAVE_CONF_QHULL_CPPFLAGS },
-    { false, "QHULL_LDFLAGS", OCTAVE_CONF_QHULL_LDFLAGS },
-    { false, "QHULL_LIBS", OCTAVE_CONF_QHULL_LIBS },
-    { false, "QRUPDATE_CPPFLAGS", OCTAVE_CONF_QRUPDATE_CPPFLAGS },
-    { false, "QRUPDATE_LDFLAGS", OCTAVE_CONF_QRUPDATE_LDFLAGS },
-    { false, "QRUPDATE_LIBS", OCTAVE_CONF_QRUPDATE_LIBS },
-    { false, "QT_CPPFLAGS", OCTAVE_CONF_QT_CPPFLAGS },
-    { false, "QT_LDFLAGS", OCTAVE_CONF_QT_LDFLAGS },
-    { false, "QT_LIBS", OCTAVE_CONF_QT_LIBS },
-    { false, "RANLIB", OCTAVE_CONF_RANLIB },
-    { false, "RDYNAMIC_FLAG", OCTAVE_CONF_RDYNAMIC_FLAG },
-    { false, "READLINE_LIBS", OCTAVE_CONF_READLINE_LIBS },
-    { false, "SED", OCTAVE_CONF_SED },
-    { false, "SHARED_LIBS", OCTAVE_CONF_SHARED_LIBS },
-    { false, "SHLEXT", OCTAVE_CONF_SHLEXT },
-    { false, "SHLEXT_VER", OCTAVE_CONF_SHLEXT_VER },
-    { false, "SH_LD", OCTAVE_CONF_SH_LD },
-    { false, "SH_LDFLAGS", OCTAVE_CONF_SH_LDFLAGS },
-    { false, "SONAME_FLAGS", OCTAVE_CONF_SONAME_FLAGS },
-    { false, "STATIC_LIBS", OCTAVE_CONF_STATIC_LIBS },
-    { false, "TERM_LIBS", OCTAVE_CONF_TERM_LIBS },
-    { false, "UMFPACK_CPPFLAGS", OCTAVE_CONF_UMFPACK_CPPFLAGS },
-    { false, "UMFPACK_LDFLAGS", OCTAVE_CONF_UMFPACK_LDFLAGS },
-    { false, "UMFPACK_LIBS", OCTAVE_CONF_UMFPACK_LIBS },
-    { false, "USE_64_BIT_IDX_T", OCTAVE_CONF_ENABLE_64 },
-    { false, "WARN_CFLAGS", OCTAVE_CONF_WARN_CFLAGS },
-    { false, "WARN_CXXFLAGS", OCTAVE_CONF_WARN_CXXFLAGS },
-    { false, "X11_INCFLAGS", OCTAVE_CONF_X11_INCFLAGS },
-    { false, "X11_LIBS", OCTAVE_CONF_X11_LIBS },
-    { false, "XTRA_CFLAGS", OCTAVE_CONF_XTRA_CFLAGS },
-    { false, "XTRA_CXXFLAGS", OCTAVE_CONF_XTRA_CXXFLAGS },
-    { false, "YACC", OCTAVE_CONF_YACC },
-    { false, "YFLAGS", OCTAVE_CONF_YFLAGS },
-    { false, "Z_CPPFLAGS", OCTAVE_CONF_Z_CPPFLAGS },
-    { false, "Z_LDFLAGS", OCTAVE_CONF_Z_LDFLAGS },
-    { false, "Z_LIBS", OCTAVE_CONF_Z_LIBS },
+    { false, "DEFS", octave::config::DEFS },
+    { false, "DL_LD", octave::config::DL_LD },
+    { false, "DL_LDFLAGS", octave::config::DL_LDFLAGS },
+    { false, "DL_LIBS", octave::config::DL_LIBS },
+    { false, "GCC_VERSION", octave::config::GCC_VERSION },
+    { false, "GXX_VERSION", octave::config::GXX_VERSION },
+    { false, "EXEEXT", octave::config::EXEEXT },
+    { false, "F77", octave::config::F77 },
+    { false, "F77_FLOAT_STORE_FLAG", octave::config::F77_FLOAT_STORE_FLAG },
+    { false, "F77_INTEGER_8_FLAG", octave::config::F77_INTEGER_8_FLAG },
+    { false, "FFLAGS", octave::config::FFLAGS },
+    { false, "FFTW3_CPPFLAGS", octave::config::FFTW3_CPPFLAGS },
+    { false, "FFTW3_LDFLAGS", octave::config::FFTW3_LDFLAGS },
+    { false, "FFTW3_LIBS", octave::config::FFTW3_LIBS },
+    { false, "FFTW3F_CPPFLAGS", octave::config::FFTW3F_CPPFLAGS },
+    { false, "FFTW3F_LDFLAGS", octave::config::FFTW3F_LDFLAGS },
+    { false, "FFTW3F_LIBS", octave::config::FFTW3F_LIBS },
+    { false, "FLIBS", octave::config::FLIBS },
+    { false, "FLTK_CPPFLAGS", octave::config::FLTK_CPPFLAGS },
+    { false, "FLTK_LDFLAGS", octave::config::FLTK_LDFLAGS },
+    { false, "FLTK_LIBS", octave::config::FLTK_LIBS },
+    { false, "FONTCONFIG_CPPFLAGS", octave::config::FONTCONFIG_CPPFLAGS },
+    { false, "FONTCONFIG_LIBS", octave::config::FONTCONFIG_LIBS },
+    { false, "FPICFLAG", octave::config::FPICFLAG },
+    { false, "FT2_CPPFLAGS", octave::config::FT2_CPPFLAGS },
+    { false, "FT2_LIBS", octave::config::FT2_LIBS },
+    { false, "GLPK_CPPFLAGS", octave::config::GLPK_CPPFLAGS },
+    { false, "GLPK_LDFLAGS", octave::config::GLPK_LDFLAGS },
+    { false, "GLPK_LIBS", octave::config::GLPK_LIBS },
+    { false, "GNUPLOT", octave::config::GNUPLOT },
+    { false, "HDF5_CPPFLAGS", octave::config::HDF5_CPPFLAGS },
+    { false, "HDF5_LDFLAGS", octave::config::HDF5_LDFLAGS },
+    { false, "HDF5_LIBS", octave::config::HDF5_LIBS },
+    { false, "LAPACK_LIBS", octave::config::LAPACK_LIBS },
+    { false, "LDFLAGS", octave::config::LDFLAGS },
+    { false, "LD_CXX", octave::config::LD_CXX },
+    { false, "LD_STATIC_FLAG", octave::config::LD_STATIC_FLAG },
+    { false, "LEX", octave::config::LEX },
+    { false, "LEXLIB", octave::config::LEXLIB },
+    { false, "LFLAGS", octave::config::LFLAGS },
+    { false, "LIBEXT", octave::config::LIBEXT },
+    { false, "LIBFLAGS", octave::config::LIBFLAGS },
+    { false, "LIBOCTAVE", octave::config::LIBOCTAVE },
+    { false, "LIBOCTINTERP", octave::config::LIBOCTINTERP },
+    { false, "LIBS", octave::config::LIBS },
+    { false, "LLVM_CPPFLAGS", octave::config::LLVM_CPPFLAGS },
+    { false, "LLVM_LDFLAGS", octave::config::LLVM_LDFLAGS },
+    { false, "LLVM_LIBS", octave::config::LLVM_LIBS },
+    { false, "LN_S", octave::config::LN_S },
+    { false, "MAGICK_CPPFLAGS", octave::config::MAGICK_CPPFLAGS },
+    { false, "MAGICK_LDFLAGS", octave::config::MAGICK_LDFLAGS },
+    { false, "MAGICK_LIBS", octave::config::MAGICK_LIBS },
+    { false, "MKOCTFILE_DL_LDFLAGS", octave::config::MKOCTFILE_DL_LDFLAGS },
+    { false, "OCTAVE_LINK_DEPS", octave::config::OCTAVE_LINK_DEPS },
+    { false, "OCTAVE_LINK_OPTS", octave::config::OCTAVE_LINK_OPTS },
+    { false, "OCT_LINK_DEPS", octave::config::OCT_LINK_DEPS },
+    { false, "OCT_LINK_OPTS", octave::config::OCT_LINK_OPTS },
+    { false, "OPENGL_LIBS", octave::config::OPENGL_LIBS },
+    { false, "OSMESA_CPPFLAGS", octave::config::OSMESA_CPPFLAGS },
+    { false, "OSMESA_LDFLAGS", octave::config::OSMESA_LDFLAGS },
+    { false, "OSMESA_LIBS", octave::config::OSMESA_LIBS },
+    { false, "PCRE_CPPFLAGS", octave::config::PCRE_CPPFLAGS },
+    { false, "PCRE_LIBS", octave::config::PCRE_LIBS },
+    { false, "PTHREAD_CFLAGS", octave::config::PTHREAD_CFLAGS },
+    { false, "PTHREAD_LIBS", octave::config::PTHREAD_LIBS },
+    { false, "QHULL_CPPFLAGS", octave::config::QHULL_CPPFLAGS },
+    { false, "QHULL_LDFLAGS", octave::config::QHULL_LDFLAGS },
+    { false, "QHULL_LIBS", octave::config::QHULL_LIBS },
+    { false, "QRUPDATE_CPPFLAGS", octave::config::QRUPDATE_CPPFLAGS },
+    { false, "QRUPDATE_LDFLAGS", octave::config::QRUPDATE_LDFLAGS },
+    { false, "QRUPDATE_LIBS", octave::config::QRUPDATE_LIBS },
+    { false, "QT_CPPFLAGS", octave::config::QT_CPPFLAGS },
+    { false, "QT_LDFLAGS", octave::config::QT_LDFLAGS },
+    { false, "QT_LIBS", octave::config::QT_LIBS },
+    { false, "RANLIB", octave::config::RANLIB },
+    { false, "RDYNAMIC_FLAG", octave::config::RDYNAMIC_FLAG },
+    { false, "READLINE_LIBS", octave::config::READLINE_LIBS },
+    { false, "SED", octave::config::SED },
+    { false, "SHARED_LIBS", octave::config::SHARED_LIBS },
+    { false, "SHLEXT", octave::config::SHLEXT },
+    { false, "SHLEXT_VER", octave::config::SHLEXT_VER },
+    { false, "SH_LD", octave::config::SH_LD },
+    { false, "SH_LDFLAGS", octave::config::SH_LDFLAGS },
+    { false, "SONAME_FLAGS", octave::config::SONAME_FLAGS },
+    { false, "STATIC_LIBS", octave::config::STATIC_LIBS },
+    { false, "TERM_LIBS", octave::config::TERM_LIBS },
+    { false, "UMFPACK_CPPFLAGS", octave::config::UMFPACK_CPPFLAGS },
+    { false, "UMFPACK_LDFLAGS", octave::config::UMFPACK_LDFLAGS },
+    { false, "UMFPACK_LIBS", octave::config::UMFPACK_LIBS },
+    { false, "WARN_CFLAGS", octave::config::WARN_CFLAGS },
+    { false, "WARN_CXXFLAGS", octave::config::WARN_CXXFLAGS },
+    { false, "X11_INCFLAGS", octave::config::X11_INCFLAGS },
+    { false, "X11_LIBS", octave::config::X11_LIBS },
+    { false, "XTRA_CFLAGS", octave::config::XTRA_CFLAGS },
+    { false, "XTRA_CXXFLAGS", octave::config::XTRA_CXXFLAGS },
+    { false, "YACC", octave::config::YACC },
+    { false, "YFLAGS", octave::config::YFLAGS },
+    { false, "Z_CPPFLAGS", octave::config::Z_CPPFLAGS },
+    { false, "Z_LDFLAGS", octave::config::Z_LDFLAGS },
+    { false, "Z_LIBS", octave::config::Z_LIBS },
     { false, "api_version", OCTAVE_API_VERSION },
     { true, "archlibdir", OCTAVE_ARCHLIBDIR },
     { true, "bindir", OCTAVE_BINDIR },
     { false, "canonical_host_type", OCTAVE_CANONICAL_HOST_TYPE },
-    { false, "config_opts", OCTAVE_CONF_config_opts },
+    { false, "config_opts", octave::config::config_opts },
     { true, "datadir", OCTAVE_DATADIR },
     { true, "datarootdir", OCTAVE_DATAROOTDIR },
     { true, "exec_prefix", OCTAVE_EXEC_PREFIX },
     { true, "fcnfiledir", OCTAVE_FCNFILEDIR },
     { true, "imagedir", OCTAVE_IMAGEDIR },
     { true, "includedir", OCTAVE_INCLUDEDIR },
     { true, "infodir", OCTAVE_INFODIR },
     { true, "infofile", OCTAVE_INFOFILE },
diff --git a/libinterp/dldfcn/__init_gnuplot__.cc b/libinterp/dldfcn/__init_gnuplot__.cc
--- a/libinterp/dldfcn/__init_gnuplot__.cc
+++ b/libinterp/dldfcn/__init_gnuplot__.cc
@@ -165,17 +165,17 @@ private:
           }
       }
   }
 };
 
 static bool
 have_gnuplot_binary (void)
 {
-  const std::string exeext = std::string (OCTAVE_CONF_EXEEXT);
+  const std::string exeext = octave::config::EXEEXT;
   const std::string path = octave_env::getenv ("PATH");
 
   octave_value_list tmp = feval ("gnuplot_binary", octave_value_list ());
   std::string gnuplot_binary = tmp(0).string_value ();
 
   string_vector args (gnuplot_binary);
   std::string gnuplot_path = search_path_for_file (path, args);
 
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -32,25 +32,25 @@ octlib_LTLIBRARIES += libinterp/liboctin
 ## files that are preprocessed to make $(DEF_FILES).  This ensures that
 ## files in $(BUILT_SOURCES) are built in the right dependency order.
 GENERATED_MAKE_BUILTINS_INCS = \
   libinterp/corefcn/defaults.h \
   libinterp/corefcn/graphics-props.cc \
   libinterp/corefcn/graphics.h \
   libinterp/corefcn/mxarray.h \
   libinterp/oct-conf-features.h \
-  libinterp/oct-conf.h \
   libinterp/parse-tree/oct-gperf.h \
   libinterp/parse-tree/oct-parse.h \
   libinterp/version.h
 
 BUILT_SOURCES += \
   $(GENERATED_MAKE_BUILTINS_INCS) \
   libinterp/builtin-defun-decls.h \
   libinterp/builtins.cc \
+  libinterp/oct-conf.cc \
   libinterp/corefcn/oct-errno.cc \
   libinterp/corefcn/oct-tex-lexer.cc \
   libinterp/corefcn/oct-tex-parser.cc \
   libinterp/corefcn/oct-tex-symbols.cc \
   libinterp/operators/ops.cc \
   libinterp/parse-tree/lex.cc \
   libinterp/parse-tree/oct-parse.cc
 
@@ -75,17 +75,17 @@ LIBINTERP_BUILT_DISTFILES = \
 ## BUT not distributed in tarball.
 LIBINTERP_BUILT_NODISTFILES = \
   libinterp/corefcn/mxarray.h \
   libinterp/corefcn/oct-errno.cc \
   libinterp/corefcn/defaults.h \
   libinterp/corefcn/graphics.h \
   libinterp/builtin-defun-decls.h \
   libinterp/operators/ops.cc \
-  libinterp/oct-conf.h \
+  libinterp/oct-conf.cc \
   libinterp/oct-conf-features.h \
   libinterp/version.h \
   $(OPT_HANDLERS) \
   $(ALL_DEF_FILES) \
   libinterp/builtins.cc
 
 libinterp_EXTRA_DIST += \
   libinterp/DOCSTRINGS \
@@ -93,38 +93,37 @@ libinterp_EXTRA_DIST += \
   libinterp/find-defun-files.sh \
   libinterp/gendoc.pl \
   libinterp/genprops.awk \
   libinterp/mk-errno-list \
   libinterp/mk-pkg-add \
   libinterp/mkbuiltins \
   libinterp/mkdefs \
   libinterp/mkops \
-  libinterp/oct-conf.in.h \
   libinterp/version.in.h \
   $(LIBINTERP_BUILT_DISTFILES)
 
 octinclude_HEADERS += \
   libinterp/corefcn/graphics-props.cc \
   libinterp/parse-tree/oct-gperf.h \
   libinterp/builtins.h \
   libinterp/builtin-defun-decls.h \
+  libinterp/oct-conf.h \
   libinterp/octave.h \
   libinterp/options-usage.h \
   $(OCTAVE_VALUE_INC) \
   $(PARSE_TREE_INC) \
   $(PARSER_INC) \
   $(LIBINTERP_OPERATORS_INC) \
   $(COREFCN_INC)
 
 nodist_octinclude_HEADERS += \
   libinterp/corefcn/mxarray.h \
   libinterp/corefcn/defaults.h \
   libinterp/corefcn/graphics.h \
-  libinterp/oct-conf.h \
   libinterp/version.h
 
 DIST_SRC += \
   libinterp/octave.cc \
   libinterp/version.cc \
   $(OCTAVE_VALUE_SRC) \
   $(PARSE_TREE_SRC) \
   $(COREFCN_SRC)
@@ -156,17 +155,17 @@ libinterp_liboctinterp_la_SOURCES = \
 nodist_libinterp_liboctinterp_la_SOURCES = \
   libinterp/corefcn/mxarray.h \
   libinterp/corefcn/oct-errno.cc \
   libinterp/corefcn/defaults.h \
   libinterp/corefcn/graphics.h \
   libinterp/operators/ops.cc \
   libinterp/builtin-defun-decls.h \
   libinterp/builtins.cc \
-  libinterp/oct-conf.h \
+  libinterp/oct-conf.cc \
   libinterp/oct-conf-features.h \
   libinterp/version.h
 
 libinterp_liboctinterp_la_LIBADD = \
   libinterp/octave-value/liboctave-value.la \
   libinterp/parse-tree/libparse-tree.la \
   libinterp/corefcn/libcorefcn.la \
   $(top_builddir)/liboctave/liboctave.la \
@@ -241,21 +240,21 @@ nobase_libinterptests_DATA = $(LIBINTERP
 ## though we don't want it.  It would be super awesome if automake
 ## would allow users to choose the header file extension.
 .yy.cc:
 	$(AM_V_YACC)$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
 
 ## Special rules:
 ## Mostly for sources which must be built before rest of compilation.
 
-## oct-conf.h must depend on Makefile.
+## oct-conf.cc must depend on Makefile.
 ## Calling configure may change default/config values.
 ## However, calling configure will also regenerate the Makefiles from
 ## Makefile.am and trigger the rules below.
-libinterp/oct-conf.h: libinterp/oct-conf.in.h Makefile
+libinterp/oct-conf.cc: libinterp/oct-conf.in.cc Makefile
 	$(AM_V_GEN)$(do_subst_config_vals)
 
 libinterp/oct-conf-features.h: $(top_builddir)/config.h libinterp/config-features.sh
 	$(AM_V_GEN)rm -f $@-t && \
 	$(srcdir)/libinterp/config-features.sh $< > $@-t && \
 	$(simple_move_if_change_rule)
 
 libinterp/version.h: libinterp/version.in.h Makefile
diff --git a/libinterp/oct-conf.h b/libinterp/oct-conf.h
new file mode 100644
--- /dev/null
+++ b/libinterp/oct-conf.h
@@ -0,0 +1,179 @@
+/*
+
+Copyright (C) 1996-2015 John W. Eaton
+
+This file is part of Octave.
+
+Octave is free software; you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 3 of the License, or (at your
+option) any later version.
+
+Octave is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with Octave; see the file COPYING.  If not, see
+<http://www.gnu.org/licenses/>.
+
+*/
+
+#if ! defined (octave_conf_h)
+#define octave_conf_h 1
+
+namespace octave
+{
+  namespace config
+  {
+    extern const char *ALL_CFLAGS;
+    extern const char *ALL_CXXFLAGS;
+    extern const char *ALL_FFLAGS;
+    extern const char *ALL_LDFLAGS;
+    extern const char *AMD_CPPFLAGS;
+    extern const char *AMD_LDFLAGS;
+    extern const char *AMD_LIBS;
+    extern const char *ARFLAGS;
+    extern const char *AR;
+    extern const char *ARPACK_CPPFLAGS;
+    extern const char *ARPACK_LDFLAGS;
+    extern const char *ARPACK_LIBS;
+    extern const char *BLAS_LIBS;
+    extern const char *CAMD_CPPFLAGS;
+    extern const char *CAMD_LDFLAGS;
+    extern const char *CAMD_LIBS;
+    extern const char *CARBON_LIBS;
+    extern const char *CC;
+    extern const char *CCOLAMD_CPPFLAGS;
+    extern const char *CCOLAMD_LDFLAGS;
+    extern const char *CCOLAMD_LIBS;
+    extern const char *CFLAGS;
+    extern const char *CHOLMOD_CPPFLAGS;
+    extern const char *CHOLMOD_LDFLAGS;
+    extern const char *CHOLMOD_LIBS;
+    extern const char *COLAMD_CPPFLAGS;
+    extern const char *COLAMD_LDFLAGS;
+    extern const char *COLAMD_LIBS;
+    extern const char *CPICFLAG;
+    extern const char *CPPFLAGS;
+    extern const char *CURL_CPPFLAGS;
+    extern const char *CURL_LDFLAGS;
+    extern const char *CURL_LIBS;
+    extern const char *CXSPARSE_CPPFLAGS;
+    extern const char *CXSPARSE_LDFLAGS;
+    extern const char *CXSPARSE_LIBS;
+    extern const char *CXXCPP;
+    extern const char *CXXFLAGS;
+    extern const char *CXXPICFLAG;
+    extern const char *CXX;
+    extern const char *DEFAULT_PAGER;
+    extern const char *DEFS;
+    extern const char *DL_LD;
+    extern const char *DL_LDFLAGS;
+    extern const char *DL_LIBS;
+    extern const char *EXEEXT;
+    extern const char *GCC_VERSION;
+    extern const char *GXX_VERSION;
+    extern const char *F77;
+    extern const char *F77_FLOAT_STORE_FLAG;
+    extern const char *F77_INTEGER_8_FLAG;
+    extern const char *FFLAGS;
+    extern const char *FFTW3_CPPFLAGS;
+    extern const char *FFTW3_LDFLAGS;
+    extern const char *FFTW3_LIBS;
+    extern const char *FFTW3F_CPPFLAGS;
+    extern const char *FFTW3F_LDFLAGS;
+    extern const char *FFTW3F_LIBS;
+    extern const char *FLIBS;
+    extern const char *FLTK_CPPFLAGS;
+    extern const char *FLTK_LDFLAGS;
+    extern const char *FLTK_LIBS;
+    extern const char *FONTCONFIG_CPPFLAGS;
+    extern const char *FONTCONFIG_LIBS;
+    extern const char *FPICFLAG;
+    extern const char *FT2_CPPFLAGS;
+    extern const char *FT2_LIBS;
+    extern const char *GLPK_CPPFLAGS;
+    extern const char *GLPK_LDFLAGS;
+    extern const char *GLPK_LIBS;
+    extern const char *GNUPLOT;
+    extern const char *HDF5_CPPFLAGS;
+    extern const char *HDF5_LDFLAGS;
+    extern const char *HDF5_LIBS;
+    extern const char *INCLUDEDIR;
+    extern const char *LAPACK_LIBS;
+    extern const char *LDFLAGS;
+    extern const char *LD_CXX;
+    extern const char *LD_STATIC_FLAG;
+    extern const char *LEXLIB;
+    extern const char *LEX;
+    extern const char *LFLAGS;
+    extern const char *LIBEXT;
+    extern const char *LIBFLAGS;
+    extern const char *LIBOCTAVE;
+    extern const char *LIBOCTINTERP;
+    extern const char *LIBS;
+    extern const char *LN_S;
+    extern const char *MAGICK_CPPFLAGS;
+    extern const char *MAGICK_LDFLAGS;
+    extern const char *MAGICK_LIBS;
+    extern const char *LLVM_CPPFLAGS;
+    extern const char *LLVM_LDFLAGS;
+    extern const char *LLVM_LIBS;
+    extern const char *MKOCTFILE_DL_LDFLAGS;
+    extern const char *OCTAVE_LINK_DEPS;
+    extern const char *OCTAVE_LINK_OPTS;
+    extern const char *OCTINCLUDEDIR;
+    extern const char *OCTLIBDIR;
+    extern const char *OCT_LINK_DEPS;
+    extern const char *OCT_LINK_OPTS;
+    extern const char *OPENGL_LIBS;
+    extern const char *OSMESA_CPPFLAGS;
+    extern const char *OSMESA_LDFLAGS;
+    extern const char *OSMESA_LIBS;
+    extern const char *PCRE_CPPFLAGS;
+    extern const char *PCRE_LIBS;
+    extern const char *PREFIX;
+    extern const char *PTHREAD_CFLAGS;
+    extern const char *PTHREAD_LIBS;
+    extern const char *QHULL_CPPFLAGS;
+    extern const char *QHULL_LDFLAGS;
+    extern const char *QHULL_LIBS;
+    extern const char *QRUPDATE_CPPFLAGS;
+    extern const char *QRUPDATE_LDFLAGS;
+    extern const char *QRUPDATE_LIBS;
+    extern const char *QT_CPPFLAGS;
+    extern const char *QT_LDFLAGS;
+    extern const char *QT_LIBS;
+    extern const char *RANLIB;
+    extern const char *RDYNAMIC_FLAG;
+    extern const char *READLINE_LIBS;
+    extern const char *SED;
+    extern const char *SHARED_LIBS;
+    extern const char *SHLEXT;
+    extern const char *SHLEXT_VER;
+    extern const char *SH_LD;
+    extern const char *SH_LDFLAGS;
+    extern const char *SONAME_FLAGS;
+    extern const char *STATIC_LIBS;
+    extern const char *TERM_LIBS;
+    extern const char *UMFPACK_CPPFLAGS;
+    extern const char *UMFPACK_LDFLAGS;
+    extern const char *UMFPACK_LIBS;
+    extern const char *WARN_CFLAGS;
+    extern const char *WARN_CXXFLAGS;
+    extern const char *X11_INCFLAGS;
+    extern const char *X11_LIBS;
+    extern const char *XTRA_CFLAGS;
+    extern const char *XTRA_CXXFLAGS;
+    extern const char *YACC;
+    extern const char *YFLAGS;
+    extern const char *Z_CPPFLAGS;
+    extern const char *Z_LDFLAGS;
+    extern const char *Z_LIBS;
+    extern const char *config_opts;
+  };
+};
+
+#endif
diff --git a/libinterp/oct-conf.in.h b/libinterp/oct-conf.in.cc
rename from libinterp/oct-conf.in.h
rename to libinterp/oct-conf.in.cc
--- a/libinterp/oct-conf.in.h
+++ b/libinterp/oct-conf.in.cc
@@ -16,642 +16,311 @@ FITNESS FOR A PARTICULAR PURPOSE.  See t
 for more details.
 
 You should have received a copy of the GNU General Public License
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
-#if ! defined (octave_conf_h)
-#define octave_conf_h 1
-
-#ifndef OCTAVE_CONF_ALL_CFLAGS
-#define OCTAVE_CONF_ALL_CFLAGS %OCTAVE_CONF_ALL_CFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_ALL_CXXFLAGS
-#define OCTAVE_CONF_ALL_CXXFLAGS %OCTAVE_CONF_ALL_CXXFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_ALL_FFLAGS
-#define OCTAVE_CONF_ALL_FFLAGS %OCTAVE_CONF_ALL_FFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_ALL_LDFLAGS
-#define OCTAVE_CONF_ALL_LDFLAGS %OCTAVE_CONF_ALL_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_AMD_CPPFLAGS
-#define OCTAVE_CONF_AMD_CPPFLAGS %OCTAVE_CONF_AMD_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_AMD_LDFLAGS
-#define OCTAVE_CONF_AMD_LDFLAGS %OCTAVE_CONF_AMD_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_AMD_LIBS
-#define OCTAVE_CONF_AMD_LIBS %OCTAVE_CONF_AMD_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_ARFLAGS
-#define OCTAVE_CONF_ARFLAGS %OCTAVE_CONF_ARFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_AR
-#define OCTAVE_CONF_AR %OCTAVE_CONF_AR%
-#endif
-
-#ifndef OCTAVE_CONF_ARPACK_CPPFLAGS
-#define OCTAVE_CONF_ARPACK_CPPFLAGS %OCTAVE_CONF_ARPACK_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_ARPACK_LDFLAGS
-#define OCTAVE_CONF_ARPACK_LDFLAGS %OCTAVE_CONF_ARPACK_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_ARPACK_LIBS
-#define OCTAVE_CONF_ARPACK_LIBS %OCTAVE_CONF_ARPACK_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_BLAS_LIBS
-#define OCTAVE_CONF_BLAS_LIBS %OCTAVE_CONF_BLAS_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CAMD_CPPFLAGS
-#define OCTAVE_CONF_CAMD_CPPFLAGS %OCTAVE_CONF_CAMD_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CAMD_LDFLAGS
-#define OCTAVE_CONF_CAMD_LDFLAGS %OCTAVE_CONF_CAMD_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CAMD_LIBS
-#define OCTAVE_CONF_CAMD_LIBS %OCTAVE_CONF_CAMD_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CARBON_LIBS
-#define OCTAVE_CONF_CARBON_LIBS %OCTAVE_CONF_CARBON_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CC
-#define OCTAVE_CONF_CC %OCTAVE_CONF_CC%
-#endif
-
-// FIXME: OCTAVE_CONF_CCC_VERSION is deprecated.  Remove in version 3.12
-#ifndef OCTAVE_CONF_CC_VERSION
-#define OCTAVE_CONF_CC_VERSION %OCTAVE_CONF_CC_VERSION%
-#endif
-
-#ifndef OCTAVE_CONF_CCOLAMD_CPPFLAGS
-#define OCTAVE_CONF_CCOLAMD_CPPFLAGS %OCTAVE_CONF_CCOLAMD_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CCOLAMD_LDFLAGS
-#define OCTAVE_CONF_CCOLAMD_LDFLAGS %OCTAVE_CONF_CCOLAMD_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CCOLAMD_LIBS
-#define OCTAVE_CONF_CCOLAMD_LIBS %OCTAVE_CONF_CCOLAMD_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CFLAGS
-#define OCTAVE_CONF_CFLAGS %OCTAVE_CONF_CFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CHOLMOD_CPPFLAGS
-#define OCTAVE_CONF_CHOLMOD_CPPFLAGS %OCTAVE_CONF_CHOLMOD_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CHOLMOD_LDFLAGS
-#define OCTAVE_CONF_CHOLMOD_LDFLAGS %OCTAVE_CONF_CHOLMOD_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CHOLMOD_LIBS
-#define OCTAVE_CONF_CHOLMOD_LIBS %OCTAVE_CONF_CHOLMOD_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_COLAMD_CPPFLAGS
-#define OCTAVE_CONF_COLAMD_CPPFLAGS %OCTAVE_CONF_COLAMD_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_COLAMD_LDFLAGS
-#define OCTAVE_CONF_COLAMD_LDFLAGS %OCTAVE_CONF_COLAMD_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_COLAMD_LIBS
-#define OCTAVE_CONF_COLAMD_LIBS %OCTAVE_CONF_COLAMD_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CPICFLAG
-#define OCTAVE_CONF_CPICFLAG %OCTAVE_CONF_CPICFLAG%
-#endif
-
-#ifndef OCTAVE_CONF_CPPFLAGS
-#define OCTAVE_CONF_CPPFLAGS %OCTAVE_CONF_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CURL_CPPFLAGS
-#define OCTAVE_CONF_CURL_CPPFLAGS %OCTAVE_CONF_CURL_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CURL_LDFLAGS
-#define OCTAVE_CONF_CURL_LDFLAGS %OCTAVE_CONF_CURL_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CURL_LIBS
-#define OCTAVE_CONF_CURL_LIBS %OCTAVE_CONF_CURL_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CXSPARSE_CPPFLAGS
-#define OCTAVE_CONF_CXSPARSE_CPPFLAGS %OCTAVE_CONF_CXSPARSE_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CXSPARSE_LDFLAGS
-#define OCTAVE_CONF_CXSPARSE_LDFLAGS %OCTAVE_CONF_CXSPARSE_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CXSPARSE_LIBS
-#define OCTAVE_CONF_CXSPARSE_LIBS %OCTAVE_CONF_CXSPARSE_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_CXXCPP
-#define OCTAVE_CONF_CXXCPP %OCTAVE_CONF_CXXCPP%
-#endif
-
-#ifndef OCTAVE_CONF_CXXFLAGS
-#define OCTAVE_CONF_CXXFLAGS %OCTAVE_CONF_CXXFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_CXXPICFLAG
-#define OCTAVE_CONF_CXXPICFLAG %OCTAVE_CONF_CXXPICFLAG%
-#endif
-
-#ifndef OCTAVE_CONF_CXX
-#define OCTAVE_CONF_CXX %OCTAVE_CONF_CXX%
-#endif
-
-// FIXME: OCTAVE_CONF_CXX_VERSION is deprecated.  Remove in version 3.12
-#ifndef OCTAVE_CONF_CXX_VERSION
-#define OCTAVE_CONF_CXX_VERSION %OCTAVE_CONF_CXX_VERSION%
-#endif
-
-#ifndef OCTAVE_CONF_DEFAULT_PAGER
-#define OCTAVE_CONF_DEFAULT_PAGER %OCTAVE_CONF_DEFAULT_PAGER%
-#endif
-
-#ifndef OCTAVE_CONF_DEFS
-#define OCTAVE_CONF_DEFS %OCTAVE_CONF_DEFS%
-#endif
-
-#ifndef OCTAVE_CONF_DL_LD
-#define OCTAVE_CONF_DL_LD %OCTAVE_CONF_DL_LD%
-#endif
-
-#ifndef OCTAVE_CONF_DL_LDFLAGS
-#define OCTAVE_CONF_DL_LDFLAGS %OCTAVE_CONF_DL_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_DL_LIBS
-#define OCTAVE_CONF_DL_LIBS %OCTAVE_CONF_DL_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_64
-#define OCTAVE_CONF_ENABLE_64 %OCTAVE_CONF_ENABLE_64%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_BOUNDS_CHECK
-#define OCTAVE_CONF_ENABLE_BOUNDS_CHECK %OCTAVE_CONF_ENABLE_BOUNDS_CHECK%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_DOCS
-#define OCTAVE_CONF_ENABLE_DOCS %OCTAVE_CONF_ENABLE_DOCS%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_DYNAMIC_LINKING
-#define OCTAVE_CONF_ENABLE_DYNAMIC_LINKING %OCTAVE_CONF_ENABLE_DYNAMIC_LINKING%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_FLOAT_TRUNCATE
-#define OCTAVE_CONF_ENABLE_FLOAT_TRUNCATE %OCTAVE_CONF_ENABLE_FLOAT_TRUNCATE%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_JIT
-#define OCTAVE_CONF_ENABLE_JIT %OCTAVE_CONF_ENABLE_JIT%
-#endif
-
-#ifndef OCTAVE_CONF_ENABLE_OPENMP
-#define OCTAVE_CONF_ENABLE_OPENMP %OCTAVE_CONF_ENABLE_OPENMP%
-#endif
-
-#ifndef OCTAVE_CONF_EXEEXT
-#define OCTAVE_CONF_EXEEXT %OCTAVE_CONF_EXEEXT%
-#endif
-
-#ifndef OCTAVE_CONF_GCC_VERSION
-#define OCTAVE_CONF_GCC_VERSION %OCTAVE_CONF_GCC_VERSION%
-#endif
-
-#ifndef OCTAVE_CONF_GXX_VERSION
-#define OCTAVE_CONF_GXX_VERSION %OCTAVE_CONF_GXX_VERSION%
-#endif
-
-#ifndef OCTAVE_CONF_F77
-#define OCTAVE_CONF_F77 %OCTAVE_CONF_F77%
-#endif
-
-#ifndef OCTAVE_CONF_F77_FLOAT_STORE_FLAG
-#define OCTAVE_CONF_F77_FLOAT_STORE_FLAG %OCTAVE_CONF_F77_FLOAT_STORE_FLAG%
-#endif
-
-#ifndef OCTAVE_CONF_F77_INTEGER_8_FLAG
-#define OCTAVE_CONF_F77_INTEGER_8_FLAG %OCTAVE_CONF_F77_INTEGER_8_FLAG%
-#endif
-
-#ifndef OCTAVE_CONF_FC
-#define OCTAVE_CONF_FC %OCTAVE_CONF_FC%
-#endif
-
-#ifndef OCTAVE_CONF_FFLAGS
-#define OCTAVE_CONF_FFLAGS %OCTAVE_CONF_FFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3_CPPFLAGSS
-#define OCTAVE_CONF_FFTW3_CPPFLAGS %OCTAVE_CONF_FFTW3_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3_LDFLAGSS
-#define OCTAVE_CONF_FFTW3_LDFLAGS %OCTAVE_CONF_FFTW3_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3_LIBS
-#define OCTAVE_CONF_FFTW3_LIBS %OCTAVE_CONF_FFTW3_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3F_CPPFLAGSS
-#define OCTAVE_CONF_FFTW3F_CPPFLAGS %OCTAVE_CONF_FFTW3F_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3F_LDFLAGSS
-#define OCTAVE_CONF_FFTW3F_LDFLAGS %OCTAVE_CONF_FFTW3F_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FFTW3F_LIBS
-#define OCTAVE_CONF_FFTW3F_LIBS %OCTAVE_CONF_FFTW3F_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_FLIBS
-#define OCTAVE_CONF_FLIBS %OCTAVE_CONF_FLIBS%
-#endif
-
-#ifndef OCTAVE_CONF_FLTK_CPPFLAGS
-#define OCTAVE_CONF_FLTK_CPPFLAGS %OCTAVE_CONF_FLTK_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FLTK_LDFLAGS
-#define OCTAVE_CONF_FLTK_LDFLAGS %OCTAVE_CONF_FLTK_LDFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FLTK_LIBS
-#define OCTAVE_CONF_FLTK_LIBS %OCTAVE_CONF_FLTK_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_FONTCONFIG_CPPFLAGS
-#define OCTAVE_CONF_FONTCONFIG_CPPFLAGS %OCTAVE_CONF_FONTCONFIG_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FONTCONFIG_LIBS
-#define OCTAVE_CONF_FONTCONFIG_LIBS %OCTAVE_CONF_FONTCONFIG_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_FPICFLAG
-#define OCTAVE_CONF_FPICFLAG %OCTAVE_CONF_FPICFLAG%
-#endif
-
-#ifndef OCTAVE_CONF_FT2_CPPFLAGS
-#define OCTAVE_CONF_FT2_CPPFLAGS %OCTAVE_CONF_FT2_CPPFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_FT2_LIBS
-#define OCTAVE_CONF_FT2_LIBS %OCTAVE_CONF_FT2_LIBS%
-#endif
-
-#ifndef OCTAVE_CONF_GLPK_CPPFLAGS
-#define OCTAVE_CONF_GLPK_CPPFLAGS %OCTAVE_CONF_GLPK_CPPFLAGS%
+#ifdef HAVE_CONFIG_H
+#include <config.h>
 #endif
 
-#ifndef OCTAVE_CONF_GLPK_LDFLAGS
-#define OCTAVE_CONF_GLPK_LDFLAGS %OCTAVE_CONF_GLPK_LDFLAGS%
-#endif
+#include "oct-conf.h"
 
-#ifndef OCTAVE_CONF_GLPK_LIBS
-#define OCTAVE_CONF_GLPK_LIBS %OCTAVE_CONF_GLPK_LIBS%
-#endif
+namespace octave
+{
+  namespace config
+  {
+    const char *ALL_CFLAGS = %OCTAVE_CONF_ALL_CFLAGS%;
 
-#ifndef OCTAVE_CONF_GNUPLOT
-#define OCTAVE_CONF_GNUPLOT %OCTAVE_CONF_GNUPLOT%
-#endif
+    const char *ALL_CXXFLAGS = %OCTAVE_CONF_ALL_CXXFLAGS%;
+
+    const char *ALL_FFLAGS = %OCTAVE_CONF_ALL_FFLAGS%;
 
-#ifndef OCTAVE_CONF_HDF5_CPPFLAGS
-#define OCTAVE_CONF_HDF5_CPPFLAGS %OCTAVE_CONF_HDF5_CPPFLAGS%
-#endif
+    const char *ALL_LDFLAGS = %OCTAVE_CONF_ALL_LDFLAGS%;
+
+    const char *AMD_CPPFLAGS = %OCTAVE_CONF_AMD_CPPFLAGS%;
+
+    const char *AMD_LDFLAGS = %OCTAVE_CONF_AMD_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_HDF5_LDFLAGS
-#define OCTAVE_CONF_HDF5_LDFLAGS %OCTAVE_CONF_HDF5_LDFLAGS%
-#endif
+    const char *AMD_LIBS = %OCTAVE_CONF_AMD_LIBS%;
+
+    const char *ARFLAGS = %OCTAVE_CONF_ARFLAGS%;
 
-#ifndef OCTAVE_CONF_HDF5_LIBS
-#define OCTAVE_CONF_HDF5_LIBS %OCTAVE_CONF_HDF5_LIBS%
-#endif
+    const char *AR = %OCTAVE_CONF_AR%;
+
+    const char *ARPACK_CPPFLAGS = %OCTAVE_CONF_ARPACK_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_INCFLAGS
-#define OCTAVE_CONF_INCFLAGS %OCTAVE_CONF_INCFLAGS%
-#endif
+    const char *ARPACK_LDFLAGS = %OCTAVE_CONF_ARPACK_LDFLAGS%;
+
+    const char *ARPACK_LIBS = %OCTAVE_CONF_ARPACK_LIBS%;
 
-#ifndef OCTAVE_CONF_INCLUDEDIR
-#define OCTAVE_CONF_INCLUDEDIR %OCTAVE_CONF_INCLUDEDIR%
-#endif
+    const char *BLAS_LIBS = %OCTAVE_CONF_BLAS_LIBS%;
 
-#ifndef OCTAVE_CONF_LAPACK_LIBS
-#define OCTAVE_CONF_LAPACK_LIBS %OCTAVE_CONF_LAPACK_LIBS%
-#endif
+    const char *CAMD_CPPFLAGS = %OCTAVE_CONF_CAMD_CPPFLAGS%;
+
+    const char *CAMD_LDFLAGS = %OCTAVE_CONF_CAMD_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_LDFLAGS
-#define OCTAVE_CONF_LDFLAGS %OCTAVE_CONF_LDFLAGS%
-#endif
+    const char *CAMD_LIBS = %OCTAVE_CONF_CAMD_LIBS%;
+
+    const char *CARBON_LIBS = %OCTAVE_CONF_CARBON_LIBS%;
 
-#ifndef OCTAVE_CONF_LD_CXX
-#define OCTAVE_CONF_LD_CXX %OCTAVE_CONF_LD_CXX%
-#endif
+    const char *CC = %OCTAVE_CONF_CC%;
+
+    const char *CCOLAMD_CPPFLAGS = %OCTAVE_CONF_CCOLAMD_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_LD_STATIC_FLAG
-#define OCTAVE_CONF_LD_STATIC_FLAG %OCTAVE_CONF_LD_STATIC_FLAG%
-#endif
+    const char *CCOLAMD_LDFLAGS = %OCTAVE_CONF_CCOLAMD_LDFLAGS%;
+
+    const char *CCOLAMD_LIBS = %OCTAVE_CONF_CCOLAMD_LIBS%;
 
-#ifndef OCTAVE_CONF_LEXLIB
-#define OCTAVE_CONF_LEXLIB %OCTAVE_CONF_LEXLIB%
-#endif
+    const char *CFLAGS = %OCTAVE_CONF_CFLAGS%;
 
-#ifndef OCTAVE_CONF_LEX
-#define OCTAVE_CONF_LEX %OCTAVE_CONF_LEX%
-#endif
+    const char *CHOLMOD_CPPFLAGS = %OCTAVE_CONF_CHOLMOD_CPPFLAGS%;
+
+    const char *CHOLMOD_LDFLAGS = %OCTAVE_CONF_CHOLMOD_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_LFLAGS
-#define OCTAVE_CONF_LFLAGS %OCTAVE_CONF_LFLAGS%
-#endif
+    const char *CHOLMOD_LIBS = %OCTAVE_CONF_CHOLMOD_LIBS%;
+
+    const char *COLAMD_CPPFLAGS = %OCTAVE_CONF_COLAMD_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_LIBEXT
-#define OCTAVE_CONF_LIBEXT %OCTAVE_CONF_LIBEXT%
-#endif
+    const char *COLAMD_LDFLAGS = %OCTAVE_CONF_COLAMD_LDFLAGS%;
+
+    const char *COLAMD_LIBS = %OCTAVE_CONF_COLAMD_LIBS%;
+
+    const char *CPICFLAG = %OCTAVE_CONF_CPICFLAG%;
 
-#ifndef OCTAVE_CONF_LIBFLAGS
-#define OCTAVE_CONF_LIBFLAGS %OCTAVE_CONF_LIBFLAGS%
-#endif
+    const char *CPPFLAGS = %OCTAVE_CONF_CPPFLAGS%;
+
+    const char *CURL_CPPFLAGS = %OCTAVE_CONF_CURL_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_LIBOCTAVE
-#define OCTAVE_CONF_LIBOCTAVE %OCTAVE_CONF_LIBOCTAVE%
-#endif
+    const char *CURL_LDFLAGS = %OCTAVE_CONF_CURL_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_LIBOCTINTERP
-#define OCTAVE_CONF_LIBOCTINTERP %OCTAVE_CONF_LIBOCTINTERP%
-#endif
+    const char *CURL_LIBS = %OCTAVE_CONF_CURL_LIBS%;
+
+    const char *CXSPARSE_CPPFLAGS = %OCTAVE_CONF_CXSPARSE_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_LIBS
-#define OCTAVE_CONF_LIBS %OCTAVE_CONF_LIBS%
-#endif
+    const char *CXSPARSE_LDFLAGS = %OCTAVE_CONF_CXSPARSE_LDFLAGS%;
+
+    const char *CXSPARSE_LIBS = %OCTAVE_CONF_CXSPARSE_LIBS%;
 
-#ifndef OCTAVE_CONF_LN_S
-#define OCTAVE_CONF_LN_S %OCTAVE_CONF_LN_S%
-#endif
+    const char *CXXCPP = %OCTAVE_CONF_CXXCPP%;
+
+    const char *CXXFLAGS = %OCTAVE_CONF_CXXFLAGS%;
 
-#ifndef OCTAVE_CONF_MAGICK_CPPFLAGS
-#define OCTAVE_CONF_MAGICK_CPPFLAGS %OCTAVE_CONF_MAGICK_CPPFLAGS%
-#endif
+    const char *CXXPICFLAG = %OCTAVE_CONF_CXXPICFLAG%;
+
+    const char *CXX = %OCTAVE_CONF_CXX%;
 
-#ifndef OCTAVE_CONF_MAGICK_LDFLAGS
-#define OCTAVE_CONF_MAGICK_LDFLAGS %OCTAVE_CONF_MAGICK_LDFLAGS%
-#endif
+    const char *DEFAULT_PAGER = %OCTAVE_CONF_DEFAULT_PAGER%;
 
-#ifndef OCTAVE_CONF_MAGICK_LIBS
-#define OCTAVE_CONF_MAGICK_LIBS %OCTAVE_CONF_MAGICK_LIBS%
-#endif
+    const char *DEFS = %OCTAVE_CONF_DEFS%;
+
+    const char *DL_LD = %OCTAVE_CONF_DL_LD%;
 
-#ifndef OCTAVE_CONF_LLVM_CPPFLAGS
-#define OCTAVE_CONF_LLVM_CPPFLAGS %OCTAVE_CONF_LLVM_CPPFLAGS%
-#endif
+    const char *DL_LDFLAGS = %OCTAVE_CONF_DL_LDFLAGS%;
+
+    const char *DL_LIBS = %OCTAVE_CONF_DL_LIBS%;
 
-#ifndef OCTAVE_CONF_LLVM_LDFLAGS
-#define OCTAVE_CONF_LLVM_LDFLAGS %OCTAVE_CONF_LLVM_LDFLAGS%
-#endif
+    const char *EXEEXT = %OCTAVE_CONF_EXEEXT%;
+
+    const char *GCC_VERSION = %OCTAVE_CONF_GCC_VERSION%;
+
+    const char *GXX_VERSION = %OCTAVE_CONF_GXX_VERSION%;
 
-#ifndef OCTAVE_CONF_LLVM_LIBS
-#define OCTAVE_CONF_LLVM_LIBS %OCTAVE_CONF_LLVM_LIBS%
-#endif
+    const char *F77 = %OCTAVE_CONF_F77%;
+
+    const char *F77_FLOAT_STORE_FLAG = %OCTAVE_CONF_F77_FLOAT_STORE_FLAG%;
 
-#ifndef OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS
-#define OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS %OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS%
-#endif
+    const char *F77_INTEGER_8_FLAG = %OCTAVE_CONF_F77_INTEGER_8_FLAG%;
 
-#ifndef OCTAVE_CONF_OCTAVE_LINK_DEPS
-#define OCTAVE_CONF_OCTAVE_LINK_DEPS %OCTAVE_CONF_OCTAVE_LINK_DEPS%
-#endif
+    const char *FFLAGS = %OCTAVE_CONF_FFLAGS%;
+
+    const char *FFTW3_CPPFLAGS = %OCTAVE_CONF_FFTW3_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_OCTAVE_LINK_OPTS
-#define OCTAVE_CONF_OCTAVE_LINK_OPTS %OCTAVE_CONF_OCTAVE_LINK_OPTS%
-#endif
+    const char *FFTW3_LDFLAGS = %OCTAVE_CONF_FFTW3_LDFLAGS%;
+
+    const char *FFTW3_LIBS = %OCTAVE_CONF_FFTW3_LIBS%;
 
-#ifndef OCTAVE_CONF_OCTINCLUDEDIR
-#define OCTAVE_CONF_OCTINCLUDEDIR %OCTAVE_CONF_OCTINCLUDEDIR%
-#endif
+    const char *FFTW3F_CPPFLAGS = %OCTAVE_CONF_FFTW3F_CPPFLAGS%;
+
+    const char *FFTW3F_LDFLAGS = %OCTAVE_CONF_FFTW3F_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_OCTLIBDIR
-#define OCTAVE_CONF_OCTLIBDIR %OCTAVE_CONF_OCTLIBDIR%
-#endif
+    const char *FFTW3F_LIBS = %OCTAVE_CONF_FFTW3F_LIBS%;
+
+    const char *FLIBS = %OCTAVE_CONF_FLIBS%;
 
-#ifndef OCTAVE_CONF_OCT_LINK_DEPS
-#define OCTAVE_CONF_OCT_LINK_DEPS %OCTAVE_CONF_OCT_LINK_DEPS%
-#endif
+    const char *FLTK_CPPFLAGS = %OCTAVE_CONF_FLTK_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_OCT_LINK_OPTS
-#define OCTAVE_CONF_OCT_LINK_OPTS %OCTAVE_CONF_OCT_LINK_OPTS%
-#endif
+    const char *FLTK_LDFLAGS = %OCTAVE_CONF_FLTK_LDFLAGS%;
+
+    const char *FLTK_LIBS = %OCTAVE_CONF_FLTK_LIBS%;
 
-#ifndef OCTAVE_CONF_OPENGL_LIBS
-#define OCTAVE_CONF_OPENGL_LIBS %OCTAVE_CONF_OPENGL_LIBS%
-#endif
+    const char *FONTCONFIG_CPPFLAGS = %OCTAVE_CONF_FONTCONFIG_CPPFLAGS%;
+
+    const char *FONTCONFIG_LIBS = %OCTAVE_CONF_FONTCONFIG_LIBS%;
 
-#ifndef OCTAVE_CONF_OSMESA_CPPFLAGS
-#define OCTAVE_CONF_OSMESA_CPPFLAGS %OCTAVE_CONF_OSMESA_CPPFLAGS%
-#endif
+    const char *FPICFLAG = %OCTAVE_CONF_FPICFLAG%;
+
+    const char *FT2_CPPFLAGS = %OCTAVE_CONF_FT2_CPPFLAGS%;
+
+    const char *FT2_LIBS = %OCTAVE_CONF_FT2_LIBS%;
 
-#ifndef OCTAVE_CONF_OSMESA_LDFLAGS
-#define OCTAVE_CONF_OSMESA_LDFLAGS %OCTAVE_CONF_OSMESA_LDFLAGS%
-#endif
+    const char *GLPK_CPPFLAGS = %OCTAVE_CONF_GLPK_CPPFLAGS%;
+
+    const char *GLPK_LDFLAGS = %OCTAVE_CONF_GLPK_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_OSMESA_LIBS
-#define OCTAVE_CONF_OSMESA_LIBS %OCTAVE_CONF_OSMESA_LIBS%
-#endif
+    const char *GLPK_LIBS = %OCTAVE_CONF_GLPK_LIBS%;
 
-#ifndef OCTAVE_CONF_PCRE_CPPFLAGS
-#define OCTAVE_CONF_PCRE_CPPFLAGS %OCTAVE_CONF_PCRE_CPPFLAGS%
-#endif
+    const char *GNUPLOT = %OCTAVE_CONF_GNUPLOT%;
+
+    const char *HDF5_CPPFLAGS = %OCTAVE_CONF_HDF5_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_PCRE_LIBS
-#define OCTAVE_CONF_PCRE_LIBS %OCTAVE_CONF_PCRE_LIBS%
-#endif
+    const char *HDF5_LDFLAGS = %OCTAVE_CONF_HDF5_LDFLAGS%;
+
+    const char *HDF5_LIBS = %OCTAVE_CONF_HDF5_LIBS%;
 
-#ifndef OCTAVE_CONF_PREFIX
-#define OCTAVE_CONF_PREFIX %OCTAVE_CONF_PREFIX%
-#endif
+    const char *INCLUDEDIR = %OCTAVE_CONF_INCLUDEDIR%;
+
+    const char *LAPACK_LIBS = %OCTAVE_CONF_LAPACK_LIBS%;
 
-#ifndef OCTAVE_CONF_PTHREAD_CFLAGS
-#define OCTAVE_CONF_PTHREAD_CFLAGS %OCTAVE_CONF_PTHREAD_CFLAGS%
-#endif
+    const char *LDFLAGS = %OCTAVE_CONF_LDFLAGS%;
+
+    const char *LD_CXX = %OCTAVE_CONF_LD_CXX%;
 
-#ifndef OCTAVE_CONF_PTHREAD_LIBS
-#define OCTAVE_CONF_PTHREAD_LIBS %OCTAVE_CONF_PTHREAD_LIBS%
-#endif
+    const char *LD_STATIC_FLAG = %OCTAVE_CONF_LD_STATIC_FLAG%;
 
-#ifndef OCTAVE_CONF_QHULL_CPPFLAGS
-#define OCTAVE_CONF_QHULL_CPPFLAGS %OCTAVE_CONF_QHULL_CPPFLAGS%
-#endif
+    const char *LEXLIB = %OCTAVE_CONF_LEXLIB%;
+
+    const char *LEX = %OCTAVE_CONF_LEX%;
 
-#ifndef OCTAVE_CONF_QHULL_LDFLAGS
-#define OCTAVE_CONF_QHULL_LDFLAGS %OCTAVE_CONF_QHULL_LDFLAGS%
-#endif
+    const char *LFLAGS = %OCTAVE_CONF_LFLAGS%;
+
+    const char *LIBEXT = %OCTAVE_CONF_LIBEXT%;
 
-#ifndef OCTAVE_CONF_QHULL_LIBS
-#define OCTAVE_CONF_QHULL_LIBS %OCTAVE_CONF_QHULL_LIBS%
-#endif
+    const char *LIBFLAGS = %OCTAVE_CONF_LIBFLAGS%;
+
+    const char *LIBOCTAVE = %OCTAVE_CONF_LIBOCTAVE%;
 
-#ifndef OCTAVE_CONF_QRUPDATE_CPPFLAGS
-#define OCTAVE_CONF_QRUPDATE_CPPFLAGS %OCTAVE_CONF_QRUPDATE_CPPFLAGS%
-#endif
+    const char *LIBOCTINTERP = %OCTAVE_CONF_LIBOCTINTERP%;
+
+    const char *LIBS = %OCTAVE_CONF_LIBS%;
 
-#ifndef OCTAVE_CONF_QRUPDATE_LDFLAGS
-#define OCTAVE_CONF_QRUPDATE_LDFLAGS %OCTAVE_CONF_QRUPDATE_LDFLAGS%
-#endif
+    const char *LN_S = %OCTAVE_CONF_LN_S%;
 
-#ifndef OCTAVE_CONF_QRUPDATE_LIBS
-#define OCTAVE_CONF_QRUPDATE_LIBS %OCTAVE_CONF_QRUPDATE_LIBS%
-#endif
+    const char *MAGICK_CPPFLAGS = %OCTAVE_CONF_MAGICK_CPPFLAGS%;
+
+    const char *MAGICK_LDFLAGS = %OCTAVE_CONF_MAGICK_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_QT_CPPFLAGS
-#define OCTAVE_CONF_QT_CPPFLAGS %OCTAVE_CONF_QT_CPPFLAGS%
-#endif
+    const char *MAGICK_LIBS = %OCTAVE_CONF_MAGICK_LIBS%;
+
+    const char *LLVM_CPPFLAGS = %OCTAVE_CONF_LLVM_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_QT_LDFLAGS
-#define OCTAVE_CONF_QT_LDFLAGS %OCTAVE_CONF_QT_LDFLAGS%
-#endif
+    const char *LLVM_LDFLAGS = %OCTAVE_CONF_LLVM_LDFLAGS%;
+
+    const char *LLVM_LIBS = %OCTAVE_CONF_LLVM_LIBS%;
 
-#ifndef OCTAVE_CONF_QT_LIBS
-#define OCTAVE_CONF_QT_LIBS %OCTAVE_CONF_QT_LIBS%
-#endif
+    const char *MKOCTFILE_DL_LDFLAGS = %OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS%;
+
+    const char *OCTAVE_LINK_DEPS = %OCTAVE_CONF_OCTAVE_LINK_DEPS%;
 
-#ifndef OCTAVE_CONF_RANLIB
-#define OCTAVE_CONF_RANLIB %OCTAVE_CONF_RANLIB%
-#endif
+    const char *OCTAVE_LINK_OPTS = %OCTAVE_CONF_OCTAVE_LINK_OPTS%;
 
-#ifndef OCTAVE_CONF_RDYNAMIC_FLAG
-#define OCTAVE_CONF_RDYNAMIC_FLAG %OCTAVE_CONF_RDYNAMIC_FLAG%
-#endif
+    const char *OCTINCLUDEDIR = %OCTAVE_CONF_OCTINCLUDEDIR%;
+
+    const char *OCTLIBDIR = %OCTAVE_CONF_OCTLIBDIR%;
 
-#ifndef OCTAVE_CONF_READLINE_LIBS
-#define OCTAVE_CONF_READLINE_LIBS %OCTAVE_CONF_READLINE_LIBS%
-#endif
+    const char *OCT_LINK_DEPS = %OCTAVE_CONF_OCT_LINK_DEPS%;
+
+    const char *OCT_LINK_OPTS = %OCTAVE_CONF_OCT_LINK_OPTS%;
 
-#ifndef OCTAVE_CONF_SED
-#define OCTAVE_CONF_SED %OCTAVE_CONF_SED%
-#endif
+    const char *OPENGL_LIBS = %OCTAVE_CONF_OPENGL_LIBS%;
+
+    const char *OSMESA_CPPFLAGS = %OCTAVE_CONF_OSMESA_CPPFLAGS%;
+
+    const char *OSMESA_LDFLAGS = %OCTAVE_CONF_OSMESA_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_SHARED_LIBS
-#define OCTAVE_CONF_SHARED_LIBS %OCTAVE_CONF_SHARED_LIBS%
-#endif
+    const char *OSMESA_LIBS = %OCTAVE_CONF_OSMESA_LIBS%;
+
+    const char *PCRE_CPPFLAGS = %OCTAVE_CONF_PCRE_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_SHLEXT
-#define OCTAVE_CONF_SHLEXT %OCTAVE_CONF_SHLEXT%
-#endif
+    const char *PCRE_LIBS = %OCTAVE_CONF_PCRE_LIBS%;
 
-#ifndef OCTAVE_CONF_SHLEXT_VER
-#define OCTAVE_CONF_SHLEXT_VER %OCTAVE_CONF_SHLEXT_VER%
-#endif
+    const char *PREFIX = %OCTAVE_CONF_PREFIX%;
+
+    const char *PTHREAD_CFLAGS = %OCTAVE_CONF_PTHREAD_CFLAGS%;
 
-#ifndef OCTAVE_CONF_SH_LD
-#define OCTAVE_CONF_SH_LD %OCTAVE_CONF_SH_LD%
-#endif
+    const char *PTHREAD_LIBS = %OCTAVE_CONF_PTHREAD_LIBS%;
+
+    const char *QHULL_CPPFLAGS = %OCTAVE_CONF_QHULL_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_SH_LDFLAGS
-#define OCTAVE_CONF_SH_LDFLAGS %OCTAVE_CONF_SH_LDFLAGS%
-#endif
+    const char *QHULL_LDFLAGS = %OCTAVE_CONF_QHULL_LDFLAGS%;
+
+    const char *QHULL_LIBS = %OCTAVE_CONF_QHULL_LIBS%;
 
-#ifndef OCTAVE_CONF_SONAME_FLAGS
-#define OCTAVE_CONF_SONAME_FLAGS %OCTAVE_CONF_SONAME_FLAGS%
-#endif
+    const char *QRUPDATE_CPPFLAGS = %OCTAVE_CONF_QRUPDATE_CPPFLAGS%;
+
+    const char *QRUPDATE_LDFLAGS = %OCTAVE_CONF_QRUPDATE_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_STATIC_LIBS
-#define OCTAVE_CONF_STATIC_LIBS %OCTAVE_CONF_STATIC_LIBS%
-#endif
+    const char *QRUPDATE_LIBS = %OCTAVE_CONF_QRUPDATE_LIBS%;
 
-#ifndef OCTAVE_CONF_TERM_LIBS
-#define OCTAVE_CONF_TERM_LIBS %OCTAVE_CONF_TERM_LIBS%
-#endif
+    const char *QT_CPPFLAGS = %OCTAVE_CONF_QT_CPPFLAGS%;
+
+    const char *QT_LDFLAGS = %OCTAVE_CONF_QT_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_UMFPACK_CPPFLAGS
-#define OCTAVE_CONF_UMFPACK_CPPFLAGS %OCTAVE_CONF_UMFPACK_CPPFLAGS%
-#endif
+    const char *QT_LIBS = %OCTAVE_CONF_QT_LIBS%;
+
+    const char *RANLIB = %OCTAVE_CONF_RANLIB%;
 
-#ifndef OCTAVE_CONF_UMFPACK_LDFLAGS
-#define OCTAVE_CONF_UMFPACK_LDFLAGS %OCTAVE_CONF_UMFPACK_LDFLAGS%
-#endif
+    const char *RDYNAMIC_FLAG = %OCTAVE_CONF_RDYNAMIC_FLAG%;
+
+    const char *READLINE_LIBS = %OCTAVE_CONF_READLINE_LIBS%;
+
+    const char *SED = %OCTAVE_CONF_SED%;
 
-#ifndef OCTAVE_CONF_UMFPACK_LIBS
-#define OCTAVE_CONF_UMFPACK_LIBS %OCTAVE_CONF_UMFPACK_LIBS%
-#endif
+    const char *SHARED_LIBS = %OCTAVE_CONF_SHARED_LIBS%;
+
+    const char *SHLEXT = %OCTAVE_CONF_SHLEXT%;
 
-#ifndef OCTAVE_CONF_WARN_CFLAGS
-#define OCTAVE_CONF_WARN_CFLAGS %OCTAVE_CONF_WARN_CFLAGS%
-#endif
+    const char *SHLEXT_VER = %OCTAVE_CONF_SHLEXT_VER%;
 
-#ifndef OCTAVE_CONF_WARN_CXXFLAGS
-#define OCTAVE_CONF_WARN_CXXFLAGS %OCTAVE_CONF_WARN_CXXFLAGS%
-#endif
+    const char *SH_LD = %OCTAVE_CONF_SH_LD%;
+
+    const char *SH_LDFLAGS = %OCTAVE_CONF_SH_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_X11_INCFLAGS
-#define OCTAVE_CONF_X11_INCFLAGS %OCTAVE_CONF_X11_INCFLAGS%
-#endif
+    const char *SONAME_FLAGS = %OCTAVE_CONF_SONAME_FLAGS%;
+
+    const char *STATIC_LIBS = %OCTAVE_CONF_STATIC_LIBS%;
 
-#ifndef OCTAVE_CONF_X11_LIBS
-#define OCTAVE_CONF_X11_LIBS %OCTAVE_CONF_X11_LIBS%
-#endif
+    const char *TERM_LIBS = %OCTAVE_CONF_TERM_LIBS%;
+
+    const char *UMFPACK_CPPFLAGS = %OCTAVE_CONF_UMFPACK_CPPFLAGS%;
 
-#ifndef OCTAVE_CONF_XTRA_CFLAGS
-#define OCTAVE_CONF_XTRA_CFLAGS %OCTAVE_CONF_XTRA_CFLAGS%
-#endif
+    const char *UMFPACK_LDFLAGS = %OCTAVE_CONF_UMFPACK_LDFLAGS%;
+
+    const char *UMFPACK_LIBS = %OCTAVE_CONF_UMFPACK_LIBS%;
 
-#ifndef OCTAVE_CONF_XTRA_CXXFLAGS
-#define OCTAVE_CONF_XTRA_CXXFLAGS %OCTAVE_CONF_XTRA_CXXFLAGS%
-#endif
+    const char *WARN_CFLAGS = %OCTAVE_CONF_WARN_CFLAGS%;
 
-#ifndef OCTAVE_CONF_YACC
-#define OCTAVE_CONF_YACC %OCTAVE_CONF_YACC%
-#endif
+    const char *WARN_CXXFLAGS = %OCTAVE_CONF_WARN_CXXFLAGS%;
+
+    const char *X11_INCFLAGS = %OCTAVE_CONF_X11_INCFLAGS%;
 
-#ifndef OCTAVE_CONF_YFLAGS
-#define OCTAVE_CONF_YFLAGS %OCTAVE_CONF_YFLAGS%
-#endif
+    const char *X11_LIBS = %OCTAVE_CONF_X11_LIBS%;
+
+    const char *XTRA_CFLAGS = %OCTAVE_CONF_XTRA_CFLAGS%;
 
-#ifndef OCTAVE_CONF_Z_CPPFLAGS
-#define OCTAVE_CONF_Z_CPPFLAGS %OCTAVE_CONF_Z_CPPFLAGS%
-#endif
+    const char *XTRA_CXXFLAGS = %OCTAVE_CONF_XTRA_CXXFLAGS%;
+
+    const char *YACC = %OCTAVE_CONF_YACC%;
+
+    const char *YFLAGS = %OCTAVE_CONF_YFLAGS%;
 
-#ifndef OCTAVE_CONF_Z_LDFLAGS
-#define OCTAVE_CONF_Z_LDFLAGS %OCTAVE_CONF_Z_LDFLAGS%
-#endif
+    const char *Z_CPPFLAGS = %OCTAVE_CONF_Z_CPPFLAGS%;
+
+    const char *Z_LDFLAGS = %OCTAVE_CONF_Z_LDFLAGS%;
 
-#ifndef OCTAVE_CONF_Z_LIBS
-#define OCTAVE_CONF_Z_LIBS %OCTAVE_CONF_Z_LIBS%
-#endif
+    const char *Z_LIBS = %OCTAVE_CONF_Z_LIBS%;
 
-#ifndef OCTAVE_CONF_config_opts
-#define OCTAVE_CONF_config_opts %OCTAVE_CONF_config_opts%
-#endif
-
-#endif
+    const char *config_opts = %OCTAVE_CONF_config_opts%;
+  };
+};
